project/iterative_sorting.py

```diff
@@ -24,16 +24,13 @@ def selection_sort(arr):
 def insertion_sort(arr):
     if not arr:
         return arr
-    result = [arr[0]]
     for i in range(1, len(arr)):
-        tmp = arr[i]
-        result.append(tmp)
-        for j in range(len(result) - 1, 0, -1):
-            if result[j] < result[j - 1]:
-                result[j], result[j - 1] = result[j - 1], result[j]
-            else:
-                break
-    return result
+        j = i - 1
+        while j >= 0 and arr[i] < arr[j]:
+            arr[i], arr[j] = arr[j], arr[i]
+            i = j
+            j = i - 1
+    return arr
```

Comment:
is this running right for you? I ran it on my computer and it's not sorting correctly.

Reply:
It passes the tests in `tests.py`, what inputs is it not sorting for?